### PR TITLE
Add leak tracking and unmute Netty4ChunkedContinuationsIT

### PR DESCRIPTION
**Pull request by nicktindall**
Related to: #109866

I've had great difficulty in reproducing this failure, this change should give us more information about where the issue is

### CODE REVIEW

File: modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4ChunkedContinuationsIT.java [hunk 1]
```diff
@@ -316,13 +317,14 @@ public void onFailure(Exception exception) {
...
         final var latch = new CountDownLatch(1);
-        refs = AbstractRefCounted.of(latch::countDown);
+        refs = LeakTracker.wrap(AbstractRefCounted.of(latch::countDown));
         return () -> {
             refs.decRef();
             try {
                 safeAwait(latch);
             } finally {
                 refs = null;
+                System.gc();
```
Comment (nicktindall) [inline modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4ChunkedContinuationsIT.java:327]: I could use `ReachabilityChecker` here to make this more foolproof?

File: modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4ChunkedContinuationsIT.java [hunk 2]
```diff
@@ -315,14 +317,16 @@ public void onFailure(Exception exception) {
...
         final var latch = new CountDownLatch(1);
-        refs = AbstractRefCounted.of(latch::countDown);
+        refs = LeakTracker.wrap(reachabilityChecker.register(AbstractRefCounted.of(latch::countDown)));
         return () -> {
             refs.decRef();
             try {
                 safeAwait(latch);
             } finally {
                 refs = null;
+                reachabilityChecker.ensureUnreachable();
```
Comment (ywangd) [inline modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4ChunkedContinuationsIT.java:329]: Nit: I wonder whether we should run the check only when safeAwait fails to avoids potential failure from `ensureUnreachable` itself? So something like
```suggestion
            boolean success = false;
            try {
                safeAwait(latch);
                success = true;
            } finally {
                refs = null;
                if (success == false) {
                    reachabilityChecker.ensureUnreachable();
                }
```
?

**Review by ywangd — APPROVED** [review]
LGTM

File: modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4ChunkedContinuationsIT.java (hunk 2 above)
Comment (nicktindall) [reply modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4ChunkedContinuationsIT.java:329]: Yes good call

**Comment by DaveCTurner** [comment]
FWIW this failure did reproduce for me locally, although it's not unusual for things like this to take a few hours of iterations to catch the culprit.

**Pull request merged**